src/pseudopeople/utilities.py

```diff
@@ -75,11 +75,23 @@ def get_index_to_noise(
     else:
         # Any index can be noised for row noise
         eligible_for_noise_idx = data.index
-    to_noise_idx = randomness_stream.filter_for_probability(
-        eligible_for_noise_idx,
-        probability=noise_level,
-        additional_key=additional_key,
-    )
+
+    # As long as noise is relatively rare, it will be faster to randomly select cells to
+    # noise rather than generating a random draw for every item eligible
+    if isinstance(noise_level, float) and noise_level < 0.2:
```

Comment:
How did you settle on 0.2?

Reply:
I used `timeit` to compare the runtime of the two methods at different noise levels. I observed much faster (~20x) performance for this method with the default 1% noise level for a shard of national data. It was better but not dramatically better for large proportions, such as 0.5. It crossed over at about 0.8 and was substantially slower than the existing method as the noise level approached 100%.

Because I did not test it in all cases, and was concerned about making things slower, I "conservatively" decided to only use this method under 20%. Of course in practice that will be nearly all the time; more noise than that should occur only when people are intentionally pushing things to their breaking point.